### PR TITLE
velero: restic plugin region value required

### PR DESCRIPTION
**Pull request by knrt10**
This is required but the code was using it as optional. Also, some docs formatting changes.

Signed-off-by: knrt10 <kautilya@kinvolk.io>

### CODE REVIEW

File: docs/configuration-reference/components/velero.md [hunk 1]
```diff
@@ -60,20 +60,20 @@ component "velero" {
 
   # openebs {
   #   credentials = file("cloud-credentails-file")
-  #		provider		= "aws"
```
Comment (invidian) [inline docs/configuration-reference/components/velero.md:63]: Seems unrelated, can we commit it separately please?

File: docs/configuration-reference/components/velero.md [hunk 2]
```diff
@@ -150,7 +151,7 @@ Example:
...
 | `restic.backup_storage_location.bucket`              | Cloud storage bucket name for storing backups.                                                                              | -                                                 | string | true     |
 | `restic.backup_storage_location.name`                | Name for backup location object on the cluster.                                                                             | -                                                 | string | false    |
-| `restic.backup_storage_location.region`              | Cloud provider region for storing snapshots.                                                                                | `eu-west-1`                                       | string | false    |
+| `restic.backup_storage_location.region`              | Cloud provider region for storing snapshots.                                                                                | `eu-west-1`                                       | string | true     |
```
Comment (invidian) [inline docs/configuration-reference/components/velero.md:154]: Should this value actually be required? If we list a default value for it?

**Review by invidian — CHANGES_REQUESTED** [review]

File: docs/configuration-reference/components/velero.md (hunk 2 above)
Comment (knrt10) [reply docs/configuration-reference/components/velero.md:154]: That's what got me confused. Because it's not using the default value

File: pkg/components/velero/restic/restic.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ type BackupStorageLocation struct {
 	Provider string `hcl:"provider"`
 	Bucket   string `hcl:"bucket"`
 	Name     string `hcl:"name,optional"`
-	Region   string `hcl:"region,optional"`
```
Comment (ipochi) [inline pkg/components/velero/restic/restic.go:41]: @knrt10 

`Region` cannot be required as its only required for aws provider in the `BackupStorageLocation` but when using (GCP)[https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/main/backupstoragelocation.md] or (Azure)[https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/main/backupstoragelocation.md] it is not required.

The idea of having default seems not right to me.

**Review by ipochi — CHANGES_REQUESTED** [review]

**Comment by ipochi** [comment]
This is not needed anymore @knrt10.

Could you please closing this.
